### PR TITLE
ROX-17712: Ensure finalizer is present before applying the resources

### PR DESCRIPTION
**Pull request by ludydoo**

### CODE REVIEW

File: pkg/reconciler/reconciler.go [hunk 1]
```diff
@@ -267,7 +267,7 @@ func WithOverrideValues(overrides map[string]string) Option {
 	}
 }
 
-// WithDependentWatchesEnabled is an Option that configures whether the
```
Comment (porridge) [inline pkg/reconciler/reconciler.go]: This a noble change, but might be a PITA if we upstream this feature separately.

File: pkg/reconciler/reconciler.go [hunk 2]
```diff
@@ -617,6 +617,15 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (res ctrl.
 		return ctrl.Result{}, err
 	}
 
+	if obj.GetDeletionTimestamp() == nil && !controllerutil.ContainsFinalizer(obj, uninstallFinalizer) {
+		log.V(1).Info("Adding uninstall finalizer")
```
Comment (porridge) [inline pkg/reconciler/reconciler.go:625]: The comment on line 671 or so will need an update.

File: pkg/reconciler/reconciler.go [hunk 3]
```diff
@@ -617,6 +617,15 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (res ctrl.
...
+		obj.SetFinalizers(append(obj.GetFinalizers(), uninstallFinalizer))
+		if err := r.client.Update(ctx, obj); err != nil {
```
Comment (porridge) [inline pkg/reconciler/reconciler.go:627]: Questions:
1. do we still need the `u.Update(updater.EnsureFinalizer(uninstallFinalizer))` at the end of this file?
2. if so, a comment here on why we're not using the updater would be useful.

**Review by porridge — COMMENTED** [review]
I forgot to click save before finishing the previous review :sweat_smile:

**Review by SimonBaeumer — APPROVED** [review]

**Comment by ludydoo** [comment]
for reference: https://github.com/operator-framework/helm-operator-plugins/pull/218

**Pull request closed without merging**